core/src/main/scala/org/http4s/Message.scala

```diff
@@ -105,6 +108,11 @@ object Message {
   object Keys {
     val TrailerHeaders = AttributeKey[Task[Headers]]
   }
+
+  def redactHeaders(headers: Headers): Headers = headers.map {
```

Comment:
I put the payload headers as a `Headers` itself.  Perhaps that's a better home for this?